### PR TITLE
Add SCD30 (CO2 Sensor) Driver

### LINKED ISSUE #216: `SCD30` Driver
I wrote a driver for the Sensirion SCD30 chip.  I'm posting this issue as directed in `CONTRIBUTING.md`.  Just waiting on CI Actions to finish and then will submit pull request.

### Device description

`SCD30` is a CO2, Temperature, and Relative Humidity sensor by Sensirion.  It supports both I2C and Modbus over UART.  This library only supports I2C.

### Device information

The product information and the data sheet can be found at:

https://www.sensirion.com/fileadmin/user_upload/customers/sensirion/Dokumente/9.5_CO2/Sensirion_CO2_Sensors_SCD30_Datasheet.pdf

https://www.sensirion.com/fileadmin/user_upload/customers/sensirion/Dokumente/9.5_CO2/Sensirion_CO2_Sensors_SCD30_Interface_Description.pdf

### Purchasing the product

The device can be purchased at:

- [x] Aliexpress
- [x] Amazon
- [x] ebay
- [x] Digi-Key

The URLs where you can buy the device are:

- https://www.digikey.ca/en/products/detail/sensirion-ag/SCD30/8445334

### Other implementation

There is an Arduino driver here: https://github.com/sparkfun/SparkFun_SCD30_Arduino_Library

### PR DESCRIPTION
**Pull request by nated0g**
closes #216 .

Tested on an Espressif ESP32-Ethernet-Kit-B development board, working.

New to contributing to larger projects with Git/Github, feedback would be welcome.  Should I have rebased my commits?  I see now my commit messages are not as requested in the docs.

### CODE REVIEW

File: components/scd30/scd30.h [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+ * @param dev      Device descriptor
+ * @param t_offset Temperature offset in degrees Celsius (°C)
+ * @return         `ESP_OK` on success
+ */
+esp_err_t scd30_get_temperature_offset(i2c_dev_t *dev, float *t_offset);
+
+/**
+ * @brief Set temperature offset in ticks.
+ *
+ * Setting the temperature offset of the SCD4x inside the customer device
```
Comment (UncleRus) [inline components/scd30/scd30.h:240]: You need to update comments to match the SCD30 datasheet

File: components/scd30/scd30.c [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+esp_err_t scd30_read_firmware_version(i2c_dev_t *dev, uint16_t *firmware_version)
+{
+    CHECK_ARG(firmware_version);
+    return execute_cmd(dev, CMD_READ_FIRMWARE_VERSION, 1, NULL, 0, firmware_version, 1);
+}
+
+esp_err_t scd30_soft_reset(i2c_dev_t *dev)
+{
+    return execute_cmd(dev, CMD_SOFT_RESET, 0, NULL, 0, NULL, 0);
+}
```
Comment (UncleRus) [inline components/scd30/scd30.c:330]: Need an empty line at the end of the file

File: components/scd30/scd30.c [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+            ets_delay_us(timeout_ms * 1000);
+    }
+    if (in_data && in_words)
+        I2C_DEV_CHECK(dev, read_resp(dev, in_data, in_words));
+    I2C_DEV_GIVE_MUTEX(dev);
+
+    return ESP_OK;
+}
+
+    ///////////////////////////////////////////////////////////////////////////////
```
Comment (UncleRus) [inline components/scd30/scd30.c:165]: Please format the code in the lines below

**Comment by UncleRus** [comment]
Great contribution, thank you!

**Comment by UncleRus** [comment]
Yes, rebasing the commits after fix would be nice

**Comment by UncleRus** [comment]
You also need to add corresponding `.rst` file to `/docs/source/groups` and include link to it in `/docs/source/index.rst` and `/docs/source/chips.rst`. Also `README.md` need to be updated - please include your component and add yoursel to the contributors list.

**Comment by nated0g** [comment]
Thanks for the feedback!

I've made the changes mentioned above.  I have never rebased before.  In `CONTRIBUTING.md` it says **not** to rebase after a review.  What is the recommended course of action now?  I push the commit (new changes are all in one commit) to my branch, and the commit will show up in this PR automatically?

**Comment by UncleRus** [comment]
Yes, all commits you make to the `scd30` branch of your fork will automatically go to this pull request.

Regarding the rebase: all this is not so important. The recommendation not to do a rebase after receiving a review is necessary so that these reviews are still linked to the corresponding source lines in the pull request. In this case, it doesn't matter, so you don't have to do the rebase at all, just push your changes.

**Comment by UncleRus** [comment]
Thank you!

**Pull request merged**